packages/examples/cvat/exchange-oracle/src/handlers/job_creation.py

```diff
@@ -2347,33 +2460,87 @@ def _job_params_label_key(ts):
+                roi_info.original_image_id: self._roi_filenames[roi_info.bbox_id]
+                for roi_info in self._roi_infos
+            }
+
+            for skeleton_label_id, skeleton_label_tasks in tasks_by_skeleton_label.items():
+                skeleton_label_filenames: list[list[str]] = []
+                gt_skeleton_label_filenames: list[list[str]] = []
+
+                for skeleton_label_task in skeleton_label_tasks:
+                    skeleton_label_filenames.append(
                         [
                             compose_data_bucket_filename(
                                 self.escrow_address, self.chain_id, self._roi_filenames[roi_id]
                             )
-                            for roi_id in skeleton_label_job.roi_ids
-                        ]
+                            for roi_id in skeleton_label_task.roi_ids
+                        ],
+                    )
+
+                    gt_skeleton_label_filenames.append(
+                        [
+                            compose_data_bucket_filename(
+                                self.escrow_address, self.chain_id, self._roi_filenames[roi_id]
+                            )
+                            for roi_id in skeleton_label_task.roi_gt_ids
+                        ],
                     )
 
                 for point_label_spec in label_specs_by_skeleton[skeleton_label_id]:
+                    point_label_name = point_label_spec["name"]
+
+                    # prepare GT dataset
+                    gt_points_dataset = dm.Dataset(
+                        categories={
+                            dm.AnnotationType.label: dm.LabelCategories.from_iterable(
+                                [point_label_name]
+                            ),
+                        },
+                        media_type=dm.Image,
+                    )
+
+                    for input_gt_filename, gt_points in self.points[
+                        (skeleton_label_id, point_label_name)
+                    ]:
+                        dataset_item_id = compose_data_bucket_filename(
+                            self.escrow_address,
+                            self.chain_id,
+                            input_filename_to_nameless[input_gt_filename].split(".", maxsplit=1)[0],
+                        )
+                        roi_info = image_id_to_roi_info[input_gt_filename]
+                        # update points coordinates in accordance with roi coordinates
+                        updated_points = [
+                            gt_points.points[0] - roi_info.roi_x,
+                            gt_points.points[1] - roi_info.roi_y,
+                        ]
+                        gt_points_dataset.put(
+                            dm.DatasetItem(
+                                id=dataset_item_id,
+                                annotations=[dm.Points(points=updated_points, label=0)],
+                            )
+                        )
```

Comment:
Consider extracting a method for this. Also, I don't think it's a bad idea to save prepared GT annotations into the oracle bucket prior to any CVAT calls. It should definitely help with debugging and traceability, in the case of task creation problems.

Reply:
done